os/net/rpl-classic/rpl-timers.c

```diff
@@ -457,11 +457,29 @@ get_probing_target(rpl_dag_t *dag)
   return probing_target;
 }
 /*---------------------------------------------------------------------------*/
+static rpl_dag_t *
+get_next_dag(rpl_instance_t *instance)
+{
+  rpl_dag_t *dag = NULL;
+  int new_dag = instance->last_dag;
+  do {
+    new_dag++;
+    if(new_dag >= RPL_MAX_DAG_PER_INSTANCE) {
+      new_dag = 0;
+    }
+    if(instance->dag_table[new_dag].used) {
+      dag = &instance->dag_table[new_dag];
+    }
+  } while(new_dag != instance->last_dag && dag == NULL);
```

Comment:
wouldn't it be a lot easier to read with a for loop over all DAGs, and a break when found?

Reply:
We have to find the next dag to probe, with a simple loop we will always return the first dag. Adding the wrap around and the check for unused dags will make the code as complex as this one (unless I'm missing something)

Reply:
right, realized what I had in mind would actually result in more iterations, so, not ideal :p Let's stick to what you have!